### PR TITLE
ENH: Add Cohen Kappa module metric to DeepMIL 

### PR DESCRIPTION
**Pull request by harshita-s**
In this PR, Cohen Kappa metric is added to DeepMIL multi-class module metrics, for PANDA DeepMIL experiments. Until now, this metric was calculated from the output CSV file.  Quadratic Weighted Kappa (QWK) used in PANDA challenge is calculated using Cohen's Kappa with quadratic weights (https://www.kaggle.com/code/reighns/understanding-the-quadratic-weighted-kappa/)

### CODE REVIEW

**Review by ajsharda17 — APPROVED** [review]

**Review by ant0nsc — APPROVED** [review]

File: hi-ml-histopathology/src/histopathology/models/deepmil.py [hunk 1]
```diff
@@ -144,6 +144,7 @@ def get_metrics(self) -> nn.ModuleDict:
                                   MetricsKey.ACC_MACRO: Accuracy(num_classes=self.n_classes, average='macro'),
                                   MetricsKey.ACC_WEIGHTED: Accuracy(num_classes=self.n_classes, average='weighted'),
                                   MetricsKey.AUROC: AUROC(num_classes=self.n_classes),
+                                  MetricsKey.COHENKAPPA: CohenKappa(num_classes=self.n_classes, weights='quadratic'),
```
Comment (dccastro) [inline hi-ml-histopathology/src/histopathology/models/deepmil.py:147]: It's not immediately obvious that this corresponds to the same quadratic-weighted kappa used in the PANDA benchmark. Because Cohen's kappa is most often used in binary classification (cf. [Wikipedia](https://en.wikipedia.org/wiki/Cohen%27s_kappa)), this can cause some confusion (it did for me!).

Consider adding some documentation (as a comment here and/or in a readme) maybe with a reference link explaining how they're related. A complementary option is to rename the key to `QUAD_KAPPA` or whatever for clarity, if you think it makes sense.

**Review by dccastro — CHANGES_REQUESTED** [review]

File: hi-ml-histopathology/testhisto/testhisto/models/test_deepmil.py [hunk 2]
```diff
@@ -125,8 +125,12 @@ def _test_lightningmodule(
         if metric_name == MetricsKey.CONF_MATRIX:
             continue
         score = metric_object(probs, bag_labels.view(batch_size,))
-        assert torch.all(score >= 0)
-        assert torch.all(score <= 1)
+        if metric_name == MetricsKey.COHENKAPPA:
+            assert torch.all(score[~score.isnan()] >= -1)
```
[the remark this answers is not in the document]
Comment (harshita-s) [reply hi-ml-histopathology/testhisto/testhisto/models/test_deepmil.py:129]: I haven't investigated when this happens, it gave a NaN in the test. This could be related to a division by zero error https://stats.stackexchange.com/questions/157832/cohens-kappa-using-irr-and-kappa2-outputs-nan

[the remark this answers is not in the document]
Comment (harshita-s) [reply hi-ml-histopathology/testhisto/testhisto/models/test_deepmil.py:129]: @dccastro where should we document this?

File: hi-ml-histopathology/src/histopathology/models/deepmil.py (hunk 1 above)
Comment (harshita-s) [reply hi-ml-histopathology/src/histopathology/models/deepmil.py:150]: I think `COHEN_KAPPA` is more general and makes more sense. I added the comment with a link of the documentation which explains specifically QWK. I also added this comment in the PR description.

**Review by dccastro — APPROVED** [review]
Great, thanks for adding the clarifications!
